ensight_lib/include/ensightbarycentriccoordinates.h

```diff
@@ -0,0 +1,254 @@
+/*
+ * Copyright (c) 2016 Fraunhofer ITWM
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+
+#ifndef ENSIGHTBARYCENTRICCOORDINATES_H
+#define ENSIGHTBARYCENTRICCOORDINATES_H
+
+#include <string>
+#include <QString>
+#include "eigentypes.h"
+
+// Forward declaration
+class EnsightCellIdentifier;
+
+/**
+ * @brief The EnsightBarycentricCoordinates class provides methods to compute barycentric coordinates in different cell types.
+ *
+ * Supported cell types are:  Ensight::Tetradhedron ,  Ensight::Pyramid ,  Ensight::Wedge ,  Ensight::Hexahedron.
+ */
+class EnsightBarycentricCoordinates
+{
+public:
+    EnsightBarycentricCoordinates();
+    EnsightBarycentricCoordinates(const Vec3 &pos, EnsightCellIdentifier* cell);
+
+    /**
+     * @brief Computes barycentric coordinates of position in cell. For 2D cells
+     * the cell is assumed to lie in the Z=0 plane. If this is not the case,
+     * computed coordinates may not be correct. Thus, for 3D geometry, only
+     * volumetric cells can be correctly represented and 2D cells should be
+     * ignored.
+     * @param pos Position
+     * @param cell Cell Identifier
+     * @param ignore2dCells ignore 2D cells. Use this for 3D geometry where only
+     * volumetric cells should be considered as containing a point.
+     */
+    bool compute(const Vec3& pos, const EnsightCellIdentifier* cell,
+                 bool ignore2dCells = true);
+
+
+    /**
+     * @brief Checks if object is a valid barycentric coordinate.
+     */
+    bool isValid();
+
+
+    /**
+     * @brief Barycentric interpolation of data
+     * @param name Field name to evaluate
+     * @return Vector of length m interpolating the data
+     */
+    Vecx evaluate(const QString& name);
```

Comment:
There are many places where 'string' and QString overloads are provided, but also many places where they are not. It'd be nice if this was more consistent (I'd learn towards just providing one and forcing the user to cast, but that's just a style thing).

Reply:
Agreed, that would have been my prefered solution too. I only added overloads because of requirements from external code.
However, there is some consistency to it (unless I've missed something): There should only be overloads for functions which are supposed to be used directly, while functions meant for internal use should provide no overlads. E.g. EnsightLib::readEnsight provides overloads, while functions in Ensight::Reader::detail:: do not.